src/crimson/common/perf_counters_service.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "common/perf_counters.h"
+#include <seastar/core/sharded.hh>
+
+namespace ceph::common {
+class PerfCountersService: public seastar::sharded<PerfCountersService>
+{
+public:
+  std::unique_ptr<PerfCountersCollectionImpl> perf_collection;
+  PerfCountersService();
+  ~PerfCountersService();
+
+  using ShardedPerfCountersService = seastar::sharded<PerfCountersService>;
+private:
+  static ShardedPerfCountersService sharded_pfservice;
+  friend PerfCountersService& local_pfservice();
+  friend ShardedPerfCountersService& sharded_pfservice();
+};
+
+inline PerfCountersService::ShardedPerfCountersService& sharded_pfservice(){
+  return PerfCountersService::sharded_pfservice;
+}
+
+inline PerfCountersService& local_pfservice() {
+  return PerfCountersService::sharded_pfservice.local();
```

Comment:
if what we care about is nothing but `perf_collection` in `PerfCountersService`, can we just expose it using a free function instead of making a public member variable?